grabbit/src/main/groovy/com/twcable/grabbit/spring/batch/repository/JcrExecutionContextDao.groovy

```diff
@@ -306,4 +303,42 @@ class JcrExecutionContextDao extends AbstractJcrDao implements ExecutionContextD
         contextAsMap.each { key, value -> context.put(key, value) }
         context
     }
+
+    @Override
+    List<String> getJobExecutionContextPaths(List<String> jobExecutionResourcePaths) {
+        JcrUtil.manageResourceResolver(resourceResolverFactory) { ResourceResolver resolver ->
+            List<String> jobExecutionContextPathsToRemove = []
+            jobExecutionResourcePaths.each { String jobExecutionResourcePath ->
+                Resource jobExecutionResource = resolver.getResource(jobExecutionResourcePath)
+                ValueMap props = jobExecutionResource.adaptTo(ValueMap)
+                Long jobExecutionId = props[JcrJobExecutionDao.EXECUTION_ID] as Long
+                String query = "select * from [nt:unstructured] as s " +
+                        "where ISDESCENDANTNODE(s,'${JOB_EXECUTION_CONTEXT_ROOT}') AND ( s.${EXECUTION_ID} = ${jobExecutionId})"
+                List<String> jobExecutionContextPaths = resolver.findResources(query, "JCR-SQL2").toList().collect { it.path }
+                jobExecutionContextPathsToRemove.addAll(jobExecutionContextPaths)
+            }
+            //Duplicates need to be removed
+            return jobExecutionContextPathsToRemove.unique() as List<String>
+        }
+    }
+
+    @Override
+    List<String> getStepExecutionContextPaths(List<String> stepExecutionResourcePaths) {
```

Comment:
I see we have the same method in JcrStepExecutionDao. I think we need a better way of dealing with these incompatible interfaces.